lib/ts/recipe/passwordless/components/features/signInAndUp/index.tsx

```diff
@@ -0,0 +1,214 @@
+            } else if (errorQueryParam === "restart_link") {
+                error = "The link you tried to click expired or was revoked. Please try again.";
+            } else if (errorQueryParam === "custom" && messageQueryParam !== null) {
+                error = messageQueryParam;
+            }
+        }
+        this.state = {
+            loaded: false,
+            loginAttemptInfo: undefined,
+            error,
+        };
+    }
+
+    getIsEmbedded = (): boolean => {
+        if (this.props.isEmbedded !== undefined) {
+            return this.props.isEmbedded;
+        }
+        return false;
+    };
+
+    getModifiedRecipeImplementation = (): RecipeInterface => {
+        return {
+            ...this.props.recipe.recipeImpl,
+            createCode: async (input) => {
+                let contactInfo;
+                if ("email" in input) {
+                    contactInfo = input.email;
+                } else {
+                    contactInfo = formatPhoneNumberIntl(input.phoneNumber);
+                }
+                const res = await this.props.recipe.recipeImpl.createCode(input);
+                if (res.status === "OK") {
+                    const loginAttemptInfo = {
+                        ...res,
+                        lastResend: new Date().getTime(),
+                        contactMethod: this.props.recipe.config.contactMethod,
+                        contactInfo,
+                    };
+                    await this.props.recipe.recipeImpl.setLoginAttemptInfo(loginAttemptInfo);
+                    this.setState((os) => ({
+                        ...os,
+                        error: undefined,
+                        loginAttemptInfo,
+                    }));
+                }
+                return res;
+            },
+            resendCode: async (input) => {
+                if (!this.state.loginAttemptInfo) {
+                    // This should never happen, but it makes TS happy
+                    throw new Error("Resend without loginAttemptInfo");
+                }
+                const res = await this.props.recipe.recipeImpl.resendCode(input);
+                if (res.status === "OK") {
+                    const loginAttemptInfo = {
+                        ...this.state.loginAttemptInfo,
+                        lastResend: new Date().getTime(),
+                    };
+                    await this.props.recipe.recipeImpl.setLoginAttemptInfo(loginAttemptInfo);
+                    this.setState((os) => ({
+                        ...os,
+                        error: undefined,
+                        loginAttemptInfo,
+                    }));
+                } else if (res.status === "RESTART_FLOW_ERROR") {
+                    await this.props.recipe.recipeImpl.clearLoginAttemptInfo();
+
+                    this.setState((os) => ({
+                        ...os,
+                        error: "Please try again.",
```

Comment:
We need to double check the copy everywhere.

Reply:
This has been changed since then, can I resolve this comment? (Maybe we could add a checkmark into the PR todos section for the UX/copy check if necessary)

Reply:
Added a checklist item to check copy